src/main/java/wolox/training/services/OpenLibrary.java

```diff
@@ -0,0 +1,78 @@
+package wolox.training.services;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import wolox.training.services.dtos.OpenLibraryBookDTO;
+
+@Service
+public class OpenLibrary {
+
+    private RestTemplate restTemplate;
+
+    @Autowired
+    private Environment env;
+
+    public OpenLibrary(RestTemplateBuilder restTemplateBuilder) {
+        restTemplate = restTemplateBuilder.build();
+    }
+
+    private JsonElement doIsbnRequest(String isbn) throws Exception {
+        String baseUrl = env.getProperty("services.openLibraryBooksBaseUrl");
+        String url = baseUrl + "/books?bibkeys=ISBN:" + isbn + "&format=json&jscmd=data";
+        URL urlPath = new URL(url);
+        HttpURLConnection con =(HttpURLConnection) urlPath.openConnection();
+        con.setRequestMethod("GET");
+        if (con.getResponseCode() != HttpURLConnection.HTTP_OK)
+            return null;
```

Comment:
Remember that it's better to use the brackets on one line ifs
[Link](https://docs.google.com/document/d/1xBoy41sG9vkrVsOd109OQ61LBQxPzZ2_F3K9e5jfHQY/edit?ts=5e4db74c#heading=h.4bnl8u3wizvm)

Besides that, this null can generate a NullPointerException then, for example in the second line from `tryGetBookByIsbn`
Maybe we can throw an exception here instead of returning null, so we can fail fast. If the external service doesn't return OK then this logic shouldn't continue.

Reply:
true, can you check my new approach?